test/integration/viewer_spec.mjs

```diff
@@ -81,10 +65,9 @@ describe("PDF viewer", () => {
             },
             [originX, originY]
           );
-          const textAfterZoomIn = await getTextAt(page, 2, originX, originY);
-          expect(textAfterZoomIn)
-            .withContext(`In ${browserName}, zoom in`)
-            .toBe(text);
+          await page.waitForFunction(
+            `document.elementFromPoint(${originX}, ${originY})?.textContent === "${text}"`
+          );
```

Comment:
> This can happen because we call `increaseScale` and then wait for the page's text layer to be visible, but it can take some time before the zoom actually occurs/completes in the viewer and in the meantime the old (pre-zoom) text layer may still be visible, causing us to continue too soon because we don't validate that we're dealing with the post-zoom text layer.

Possibly a stupid question, but isn't there a theoretical risk that these new checks could also target the "old" textLayer if the timing is just right?
Would it be even safer if we awaited the "textlayerrendered" event, before checking the textContent?